seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,48 +1,107 @@
-    #  just be the workspace name. Make sure to set workspace_namespace and workspace_name correctly;
-    # 5) Add families/individuals based on the uploaded pedigree file;
-    # 6) Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
-    #  project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
-    return create_json_response(response_json)
+    # Validate that the current user has logged in through google and has sufficient permissions
+    check_workspace_perm(request.user, CAN_EDIT, namespace, name, can_share=True)
+
+    projects = Project.objects.filter(workspace_namespace=namespace, workspace_name=name)
+    if projects:
+        error = 'Project "{}" for workspace "{}/{}" exists.'.format(projects.first().name, namespace, name)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Parse families/individuals in the uploaded pedigree file
+    json_records = load_uploaded_file(request_json['uploadedFileId'])
+    pedigree_records, errors, ped_warnings = parse_pedigree_table(json_records, 'uploaded pedigree file', user=request.user)
+    errors += ped_warnings
+    if errors:
+        return create_json_response({'errors': errors}, status=400)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    add_service_account(request.user, namespace, name)
+
+    # Create a new Project in seqr
+    project_args = {
+        'name': name,
+        'genome_version': request_json['genomeVersion'],
+        'description': request_json.get('description', ''),
+        'workspace_namespace': namespace,
+        'workspace_name': name,
+    }
+
+    project = create_model_from_json(Project, project_args, user=request.user)
+
+    # add families and individuals according to the uploaded individual records
+    _, updated_individuals = add_or_update_individuals_and_families(
+        project, individual_records=pedigree_records, user=request.user
+    )
+    individual_ids_tsv = '\n'.join([individual.individual_id for individual in updated_individuals])
+
+    # Send an email to all seqr data managers
+    try:
+        send_load_data_email(project, request.scheme+'://'+get_current_site(request).domain, individual_ids_tsv)
```

Comment:
`get_current_site ` is an antipattern, you should use the `BASE_URL` from settings the way we do in `send_load_data_email` and not pass it as an argument to this function

Reply:
Good. I was wondering about how to get the URL for a while. I did notice that `BASE_URL` in the settings but I didn't know what the value it will have in the production env. Now, I find the value in the deployment scripts.